docs/DeveloperGuide.md

```diff
@@ -1,25 +1,277 @@
+The **Sequence Diagram** above shows how the components interact for a basic `create <deck name>` command where a new deck is created and added in to the `Deck List`.
+
+### User Interface
+
+**API**: [seedu/ecardnomics/Ui.java](https://github.com/AY2021S1-CS2113-T14-2/tp/tree/master/src/main/java/seedu/ecardnomics/Ui.java)
+
+The UI contains String constants that represent the outputs
+that the application is defined to produce.
+
+The `UI` component has two main purposes:
+* Reading user input from the console.
+* Printing program output to the console.
+
+Reading of user input is done using the method `readUserInput()`
+which reads one line of user input. The other methods within `UI` are
+called when a specific output needs to be printed.
+
+The `UI` component passes the user input to the `NormalParser` and
+`DeckParser` components that will extract the relevant information.
+The `UI` component provides its printing methods to `NormalParser`
+and `DeckParser` for printing the appropriate output when required.
+
+### Logic
+#### Overall Logic
+![DG-Overall Logic UML](./images-dg/Logic-DG.png?raw=true "Overall Logic Diagram")
+
+1. The overall logic component consists of the Parser class and Command class.
+2. The Parser parses the user input and creates the respective Command object.
+3. This command will be executed by the Main class.
+4. The command execution then can affect the Model (e.g. creating a new deck)
+
+#### Commands
+
+![DG-Design Commands UML](./images-dg/DG-Design-Commands.png?raw=true "Commands UML Class Diagram")
+
+**API**: [seedu.ecardnomics/command](https://github.com/AY2021S1-CS2113-T14-2/tp/tree/master/src/main/java/seedu
+/ecardnomics/command)
+
+Commands are primarily classified into three categories, `NormalCommand`, `DeckCommand`, and `GameCommand`,
+ corresponding to the application's Normal, Deck, and Game Modes, respectively. All three are abstract children
+  derived from the overarching abstract class `Command`. The basis `Command` class is defined as such:
+   
+```java
+public abstract class Command {
+    public abstract void execute();
+}
+```
+
+It only requires that all derived children implement the `execute()` method. The only two classes not belonging to
+ individual modes are `ExitCommand` and `VoidCommand`. The former is so that users can call the command `exit` from
+  anywhere in the application, while the latter is a catch-all "command" for all erroneous commands a user enters. 
+   
+The `Parser` classes play important roles in execution of specific commands, e.g. `CreateCommand`, because
+ they define methods that check and ensure the conformity of user input to the commands' expected input. Below is a
+  sequence diagram showcasing this interaction, for execution of a `CreateCommand`, e.g. `create
+   microeconomics`:
+  
+![DG-Design CreateCommand Sequence UML](./images-dg/DG-Design-Sequence-Diagram.png?raw=true "CreateCommand UML
+ Sequence Diagram")
```

Comment:
You can consider terminating CreateCommand

![image](https://user-images.githubusercontent.com/60382274/97516099-5c274680-19cd-11eb-93e2-c6a6844b1052.png)